components/ValidatedInput.js

```diff
@@ -164,46 +166,50 @@ class ValidatedInput extends Component {
   }
 
   render () {
+    const {
+      hasBeenFocused,
+      messages,
+    } = this.state
     const classNames = [
       'validated-input',
       (this.props.className || ''),
     ]
 
     return (
-      <div className={classNames.join(' ')}>
-        <input {...this.renderProps} />
-
-        {this.renderMessages()}
+      <div
+        className={classNames.join(' ')}
+        data-t="validated-input:wrapper">
+        <input
+          data-t="validated-input:input"
+          {...this.renderProps} />
+
+        <FontAwesomeIcon
```

Comment:
Separating this from the main render function was done on purpose. PasswordInput uses the same renderMessages function to avoid disparity and duplication between the two input types.